src/App.tsx

```diff
@@ -1,26 +1,255 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
 
-const USER_ID = 0;
+import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  getTodos,
+  postTodos,
+  deleteTodos,
+  patchTodo,
+} from './api/todos';
+import { TodoList } from './Components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { Footer } from './Components/Footer/Footer';
+import { FilterStatus } from './types/FilterStatus';
+import { getFilteredTodos } from './utils/getFilteredTodos';
+import { ErrorNotification } from './Components/ErrorNotification';
+import { getRandomNumberId } from './utils/getRandomNumberId';
+import { getCountOfCompletedTodos } from './utils/getCountOfCompletedTodos';
+import { ErrorMessage } from './types/ErrorMessage';
 
 export const App: React.FC = () => {
+  const [newTitle, setNewTitle] = useState<string>('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterStatus>(FilterStatus.All);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage | string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoIdsInLoading, setTodoIdsInLoading] = useState<number[]>([]);
+  const [todoEditingId, setTodoEditingId] = useState<number | null>(null);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handlerErrorShow = (error: string): void => {
+    setErrorMessage(error);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        handlerErrorShow(ErrorMessage.LoadError);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setNewTitle('');
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isLoading]);
+
+  const handlerAddTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsLoading(true);
+    const normalizedTitle = newTitle.trim();
+
+    if (!normalizedTitle) {
+      handlerErrorShow(ErrorMessage.TitleError);
+      setIsLoading(false);
+
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getRandomNumberId(),
```

Comment:
Please re-check the API doc

I think we should not give the id to the server

Reply:
I understand that when we send our data to the API, the API put its ids for the objects. But I need these random identifiers to store my locale todos

Reply:
@ivan-gritsenko 

I believe that we can you response from `postTodos` to update local state

https://mate-academy.github.io/fe-students-api/todos.html#single